models/event.go

```diff
@@ -67,6 +68,19 @@ func (e Event) MarshalJSON() ([]byte, error) {
 	return json.Marshal(test)
 }
 
+func (e Event) MarshalCBOR() ([]byte, error) {
```

Comment:
Is this a similar function in use to MarshalJSON()? By that I mean does this implement an interface like [json.Marshaler](https://golang.org/pkg/encoding/json/#Marshaler)? If so, it shouldn't be called directly. If not, I wouldn't utilize this approach because it's conflating an existing convention.